core/src/main/java/org/apache/accumulo/core/iterators/SortedMapIterator.java

```diff
@@ -39,7 +40,7 @@
  * columnFamilies passed into seek(). If you want a Map-backed Iterator that returns only sought
  * CFs, construct a new ColumnFamilySkippingIterator(new SortedMapIterator(map)).
  *
- * @see org.apache.accumulo.core.iterators.system.ColumnFamilySkippingIterator
+ * @see ColumnFamilySkippingIterator
  */
 public class SortedMapIterator implements InterruptibleIterator {
```

Comment:
Can we implement or extent something that is not in iteratorsImpl?

Reply:
Good catch.  Yeah I'll see what I can do.

Reply:
This appears to be mainly used by tests, with the exception of the InMemoryMap.  I think we may have to split this into two Iterators, one that is interruptible for internal use and the other generic SortedMapIterator that implements SKVI for public use, which shouldn't need to be interrupted.

Reply:
> I think we may have to split this into two Iterators

I like that approach.  I wonder if it would be possible to make the system one extend/use code from the public api one.

Reply:
Yup that seems to work well.  I am running ITs to make sure I didn't break anything.

Reply:
I am little concerned about the all the casting we do with InterrubitleIterator.  Here is an example: https://github.com/apache/accumulo/blob/61b2b9ebfc815054b1b8281ff485b993823aa4a5/core/src/main/java/org/apache/accumulo/core/iteratorsImpl/system/SourceSwitchingIterator.java#L263

Reply:
What's the concern? That we're not checking instanceof properly first?

Reply:
Yeah.  I just pushed a fix to a Test in c2d342b that was failing due to the assumption that SortedMapIterator was an InterruptibleIterator.

Reply:
Making assumptions about our own code is probably okay, if we have good test coverage that can catch it, but making assumptions about user-provided code could be really bad. Hopefully, we only do the former. In any case, it probably wouldn't hurt to add some `instanceof` guards and handle the error sanely.

Reply:
ITs and unit tests are passing so I will make this a follow on ticket.

Reply:
Just looked at InterruptibleMapITerator, it looks nice